app/services/factories/event.rb

```diff
@@ -12,7 +12,7 @@ def call(event)
     private
 
       def event_namespace_for_event(event)
-        event_identifiers[event.underscore.intern].to_s
+        event_identifiers[event.underscore.intern].to_s if event
```

Comment:
Do we need this and the catch block in the caller?

Reply:
No, we don't, good spot :)